tests/torch/sparsity/movement/helpers.py

```diff
@@ -0,0 +1,737 @@
+        scheduler_keys = set(scheduler_config.__dict__)
+        algo_keys = set(algo_config.__dict__)
+        for key, value in override_kwargs.items():
+            if key in model_keys:
+                setattr(model_config, key, value)
+            elif key in algo_keys:
+                setattr(algo_config, key, value)
+            elif key in scheduler_keys:
+                setattr(scheduler_config, key, value)
+            else:
+                raise ValueError(f'Unknown config: {key}')
+        return cls(model_config, algo_config, log_dir)
+
+    @property
+    @abstractmethod
+    def model_input_info(self) -> List[dict]:
+        pass
+
+    @property
+    @abstractmethod
+    def transformer_block_info(self) -> List[TransformerBlockInfo]:
+        pass
+
+    @property
+    def model(self) -> torch.nn.Module:
+        torch_model = self._create_model()
+        g = torch.Generator()
+        g.manual_seed(42)
+        with torch.no_grad():
+            for _, parameter in torch_model.named_parameters():
+                parameter.normal_(generator=g)
+        return torch_model
+
+    @property
+    def nncf_config(self) -> NNCFConfig:
+        config_dict = {
+            'input_info': self.model_input_info,
+            'compression': self.algo_config.to_dict()}
+        if self.log_dir is not None:
+            config_dict['log_dir'] = str(self.log_dir)
+        return NNCFConfig.from_dict(config_dict)
+
+    @property
+    def scheduler_params(self):
+        return self.algo_config.scheduler_params
+
+    @staticmethod
+    @abstractmethod
+    def get_nncf_modules_in_transformer_block_order(
+            compressed_model: NNCFNetwork) -> List[TransformerBlockItemOrderedDict]:
+        pass
+
+    @abstractmethod
+    def _create_model(self) -> torch.nn.Module:
+        pass
+
+    def set_log_dir(self, log_dir=None):
+        self.log_dir = str(log_dir)
+        if log_dir is not None:
+            Path(log_dir).mkdir(exist_ok=True, parents=True)
+
+    def get(self, key: str):
+        if key == 'log_dir':
+            return self.log_dir
+        if key in self._model_keys:
+            return getattr(self.model_config, key)
+        if key in self._algo_keys:
+            return getattr(self.algo_config, key)
+        if key in self._scheduler_keys:
+            return getattr(self.algo_config.scheduler_params, key)
+        raise KeyError(f'"{key}" not found.')
+
+    def set(self, **kwargs):
+        for key, value in kwargs.items():
+            if key == 'log_dir':
+                self.set_log_dir(value)
+            elif key in self._model_keys:
+                setattr(self.model_config, key, value)
+            elif key in self._algo_keys:
+                setattr(self.algo_config, key, value)
+            elif key in self._scheduler_keys:
+                setattr(self.algo_config.scheduler_params, key, value)
+            else:
+                raise KeyError(f'"{key}" not found.')
```

Comment:
It doesn't make sense to try and save up on line length by joining all keys of the attributes of this recipe into one namespace accessible via this set/get implementation. This will make understanding of the test structure harder because one would have to guess in which of the three structs the key is defined in, and will also hide mistakes that would have been caught by static analysis. Remove these two methods, and instead get the values via fully qualified attribute access and `.`, e.g. `recipe.get('compression_lr_multiplier')` -> `recipe.algo_config.compression_lr_multiplier`.